src/sentry/eventstore/snuba/backend.py

```diff
@@ -249,6 +249,14 @@ def __get_events(
                 start = min(event.datetime for event in nodestore_events)
                 end = max(event.datetime for event in nodestore_events) + timedelta(seconds=1)
 
+                if filter.start or filter.end:
+                    if filter.start:
+                        start = max(start, filter.start)
+                    if filter.end:
+                        end = min(end, filter.end)
+                    if start > end:
+                        return []
```

Comment:
These cascading `if`s and the reliance of their ordering is error prone. This can probably be written like:
```
start = max(start, filter.start) if filter.start else start
end = min(end, filter.end) if filter.end else end
if start > end:
    return []
```

Reply:
Got it, the reason I wrote it like this is because I wanted to preserve the behavior of the `start > end` case further downstream. In case users of this function are expecting specific errors to be thrown. I.e. I only want the early termination to happen in this special case

Reply:
Rewrote it so these conditions and early term are no longer necessary!